src/elf.rs

```diff
@@ -817,16 +893,28 @@ impl<'a> Elf<'a> {
         /////////////////////////////////////
         for symbol in self.special_symbols.into_iter() {
             debug!("Special Symbol: {:?}", symbol);
-            file.iowrite_with(symbol, self.ctx)?;
+            let mut sym = symbol.clone();
+            if sym.st_shndx >= SHN_LORESERVE as usize {
+                sym.st_shndx = SHN_XINDEX as usize;
+            }
+            file.iowrite_with(sym, self.ctx)?;
         }
         for (_id, section) in self.sections.into_iter() {
             debug!("Section Symbol: {:?}", section.symbol);
-            file.iowrite_with(section.symbol, self.ctx)?;
+            let mut sym = section.symbol.clone();
+            if sym.st_shndx >= SHN_LORESERVE as usize {
+                sym.st_shndx = SHN_XINDEX as usize;
+            }
+            file.iowrite_with(sym, self.ctx)?;
             section_headers.push(section.header);
         }
         for (_id, symbol) in self.symbols.into_iter() {
             debug!("Symbol: {:?}", symbol);
-            file.iowrite_with(symbol, self.ctx)?;
+            let mut sym = symbol.clone();
+            if sym.st_shndx >= SHN_LORESERVE as usize {
+                sym.st_shndx = SHN_XINDEX as usize;
```

Comment:
Have to check the “spec” (ie some random stuff I find on the internet 🤣) but this may be causing the issue, if the SHN_XINDEX is only for section symbols, then this will point the regular symbols at the section symtab which won’t be right. May be able to remove this line